File: algorithm/02_searching/03_Back_Tracking/solve.py
yedam = []

def Fist_of_the_North_Star(start, remaining) :
    if(remaining == 0) :
        if((yedam[0] == 1) and (yedam[-1] == 1)) :
            return
        print(''.join(map(str, yedam)))
        return
        
    for i in range(start, len(yedam)) :
        yedam[i] = 1
        Fist_of_the_North_Star(i + 1, remaining - 1)
        yedam[i] = 0
    return

n, m = map(int, input().split())
for i in range(n) : 
    yedam.append(0)
Fist_of_the_North_Star(0, m)

File: algorithm/02_searching/03_Back_Tracking/solve.cpp
#include <iostream>
#include <vector>

using namespace std;
vector<int> points;

void fistOfTheNorthStar(vector<int>& arr, int start, int remaining)
{
  if(remaining == 0)
  {
    if(arr[0] == 1 && arr.back() == 1) return;
    for(int x : arr) cout << x;
    cout << "\n";
    return;
  }

  for(int i = start; i < arr.size(); i++)
  {
    arr[i] = 1;
    fistOfTheNorthStar(arr, i + 1, remaining - 1);
    arr[i] = 0;
  }
  return;
}
int main()
{
  int n, m;
  cin >> n >> m;
  vector<int> points (n, 0);
  fistOfTheNorthStar(points, 0, m);
  return 0;
}